diff --git a/src/main/scala/gitbucket/core/GitBucketCoreModule.scala b/src/main/scala/gitbucket/core/GitBucketCoreModule.scala
index ff4b726d..fab62f38 100644
--- a/src/main/scala/gitbucket/core/GitBucketCoreModule.scala
+++ b/src/main/scala/gitbucket/core/GitBucketCoreModule.scala
@@ -124,7 +124,8 @@ object GitBucketCoreModule
       new Version("4.44.0", new LiquibaseMigration("update/gitbucket-core_4.44.xml")),
       new Version("4.45.0"),
       new Version("4.46.0", new LiquibaseMigration("update/gitbucket-core_4.46.xml")),
-      new Version("4.46.1")
+      new Version("4.46.1"),
+      new Version("4.47.0", new LiquibaseMigration("update/gitbucket-core_4.47.xml"))
     ) {
   java.util.logging.Logger.getLogger("liquibase").setLevel(Level.SEVERE)
 }
diff --git a/src/main/scala/gitbucket/core/model/Repository.scala b/src/main/scala/gitbucket/core/model/Repository.scala
index 121f9be9..30ffdbfe 100644
--- a/src/main/scala/gitbucket/core/model/Repository.scala
+++ b/src/main/scala/gitbucket/core/model/Repository.scala
@@ -13,6 +13,7 @@ trait RepositoryComponent extends TemplateComponent { self: Profile =>
     val registeredDate = column[java.util.Date]("REGISTERED_DATE")
     val updatedDate = column[java.util.Date]("UPDATED_DATE")
     val lastActivityDate = column[java.util.Date]("LAST_ACTIVITY_DATE")
+    val lastSyncTime = column[java.util.Date]("LAST_SYNC_TIME")
     val originUserName = column[String]("ORIGIN_USER_NAME")
     val originRepositoryName = column[String]("ORIGIN_REPOSITORY_NAME")
     val parentUserName = column[String]("PARENT_USER_NAME")
@@ -37,6 +38,7 @@ trait RepositoryComponent extends TemplateComponent { self: Profile =>
           registeredDate,
           updatedDate,
           lastActivityDate,
+          lastSyncTime.?,
           originUserName.?,
           originRepositoryName.?,
           parentUserName.?,
@@ -67,6 +69,7 @@ trait RepositoryComponent extends TemplateComponent { self: Profile =>
             repository._10,
             repository._11,
             repository._12,
+            repository._13,
             RepositoryOptions.apply.tupled.apply(options)
           )
         },
@@ -82,6 +85,7 @@ trait RepositoryComponent extends TemplateComponent { self: Profile =>
                 r.registeredDate,
                 r.updatedDate,
                 r.lastActivityDate,
+                r.lastSyncTime,
                 r.originUserName,
                 r.originRepositoryName,
                 r.parentUserName,
@@ -108,6 +112,7 @@ case class Repository(
   registeredDate: java.util.Date,
   updatedDate: java.util.Date,
   lastActivityDate: java.util.Date,
+  lastSyncTime: Option[java.util.Date],
   originUserName: Option[String],
   originRepositoryName: Option[String],
   parentUserName: Option[String],
diff --git a/src/main/scala/gitbucket/core/service/RepositoryService.scala b/src/main/scala/gitbucket/core/service/RepositoryService.scala
index e0df0a3b..4f2481d8 100644
--- a/src/main/scala/gitbucket/core/service/RepositoryService.scala
+++ b/src/main/scala/gitbucket/core/service/RepositoryService.scala
@@ -50,6 +50,7 @@ trait RepositoryService {
         registeredDate = currentDate,
         updatedDate = currentDate,
         lastActivityDate = currentDate,
+        lastSyncTime = None,
         originUserName = originUserName,
         originRepositoryName = originRepositoryName,
         parentUserName = parentUserName,
