diff --git a/doc/authenticator.md b/doc/authenticator.md
index a51b1dbe..347e4a0d 100644
--- a/doc/authenticator.md
+++ b/doc/authenticator.md
@@ -58,3 +58,9 @@ Currently, GitBucket provides below authenticators:
 |GroupManagerAuthenticator |managersOnly     |Allows only the group managers.                                                       |
 
 Of course, if you make a new plugin, you can implement your own authenticator according to requirement in your plugin.
+
+## Authenticator Lock
+
+In high concurrency scenarios, you may encounter race conditions in authenticators. To prevent this, GitBucket provides an optional lock mechanism. You can enable it by setting `authenticator_lock_enabled=true` in gitbucket.conf.
+
+When enabled, all authenticators will use a shared lock to ensure that only one authentication process runs at a time.
diff --git a/src/main/scala/gitbucket/core/service/SystemSettingsService.scala b/src/main/scala/gitbucket/core/service/SystemSettingsService.scala
index 6ec6c8c2..b5641de5 100644
--- a/src/main/scala/gitbucket/core/service/SystemSettingsService.scala
+++ b/src/main/scala/gitbucket/core/service/SystemSettingsService.scala
@@ -32,6 +32,7 @@ trait SystemSettingsService {
     props.setProperty(Notification, settings.basicBehavior.notification.toString)
     props.setProperty(LimitVisibleRepositories, settings.basicBehavior.limitVisibleRepositories.toString)
     props.setProperty(CompareNoCheckByDefault, settings.basicBehavior.compareNoCheckByDefault.toString)
+    props.setProperty(AuthenticatorLockEnabled, settings.basicBehavior.authenticatorLockEnabled.toString)
     props.setProperty(SshEnabled, settings.ssh.enabled.toString)
     settings.ssh.bindAddress.foreach { bindAddress =>
       props.setProperty(SshBindAddressHost, bindAddress.host.trim())
@@ -132,7 +133,8 @@ trait SystemSettingsService {
         getValue(props, Gravatar, false),
         getValue(props, Notification, false),
         getValue(props, LimitVisibleRepositories, false),
-        getValue(props, CompareNoCheckByDefault, false)
+        getValue(props, CompareNoCheckByDefault, false),
+        getValue(props, AuthenticatorLockEnabled, false)
       ),
       Ssh(
         enabled = getValue(props, SshEnabled, false),
@@ -287,6 +289,7 @@ object SystemSettingsService {
     notification: Boolean,
     limitVisibleRepositories: Boolean,
     compareNoCheckByDefault: Boolean,
+    authenticatorLockEnabled: Boolean
   )
 
   case class RepositoryOperation(
@@ -421,6 +424,7 @@ object SystemSettingsService {
   private val Notification = "notification"
   private val LimitVisibleRepositories = "limitVisibleRepositories"
   private val CompareNoCheckByDefault = "compare_no_check_by_default"
+  private val AuthenticatorLockEnabled = "authenticator_lock_enabled"
   private val SshEnabled = "ssh"
   private val SshHost = "ssh.host"
   private val SshPort = "ssh.port"
diff --git a/src/main/scala/gitbucket/core/util/Authenticator.scala b/src/main/scala/gitbucket/core/util/Authenticator.scala
index ba838d27..d88fdd14 100644
--- a/src/main/scala/gitbucket/core/util/Authenticator.scala
+++ b/src/main/scala/gitbucket/core/util/Authenticator.scala
@@ -1,23 +1,42 @@
 package gitbucket.core.util
 
 import gitbucket.core.controller.ControllerBase
-import gitbucket.core.service.{AccountService, RepositoryService}
+import gitbucket.core.service.{AccountService, RepositoryService, SystemSettingsService}
 import gitbucket.core.model.Role
 import RepositoryService.RepositoryInfo
 import Implicits._
 
+/**
+ * Shared lock for authenticators to prevent race conditions in high concurrency scenarios.
+ */
+private[util] object AuthenticatorLock {
+  val lock = new Object()
+}
+
 /**
  * Allows only oneself and administrators.
  */
-trait OneselfAuthenticator { self: ControllerBase =>
+trait OneselfAuthenticator { self: ControllerBase & SystemSettingsService =>
   protected def oneselfOnly(action: => Any) = { authenticate(action) }
   protected def oneselfOnly[T](action: T => Any) = (form: T) => { authenticate(action(form)) }
 
   private def authenticate(action: => Any) = {
-    context.loginAccount match {
-      case Some(x) if x.isAdmin                      => action
-      case Some(x) if request.paths(0) == x.userName => action
-      case _                                         => Unauthorized()
+    withLockIfEnabled {
+      context.loginAccount match {
+        case Some(x) if x.isAdmin                      => action
+        case Some(x) if request.paths(0) == x.userName => action
+        case _                                         => Unauthorized()
+      }
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
+        action
+      }
+    } else {
+      action
     }
   }
 }
@@ -25,41 +44,65 @@ trait OneselfAuthenticator { self: ControllerBase =>
 /**
  * Allows only the repository owner and administrators.
  */
-trait OwnerAuthenticator { self: ControllerBase & RepositoryService & AccountService =>
+trait OwnerAuthenticator { self: ControllerBase & RepositoryService & AccountService & SystemSettingsService =>
   protected def ownerOnly(action: RepositoryInfo => Any) = { authenticate(action) }
   protected def ownerOnly[T](action: (T, RepositoryInfo) => Any) = (form: T) => { authenticate(action(form, _)) }
 
   private def authenticate(action: (RepositoryInfo) => Any) = {
-    val userName = params("owner")
-    val repoName = params("repository")
-    getRepository(userName, repoName).map { repository =>
-      context.loginAccount match {
-        case Some(x) if x.isAdmin                      => action(repository)
-        case Some(x) if repository.owner == x.userName => action(repository)
-        // TODO Repository management is allowed for only group managers?
-        case Some(x) if getGroupMembers(repository.owner).exists { m =>
-              m.userName == x.userName && m.isManager
-            } =>
-          action(repository)
-        case Some(x) if getCollaboratorUserNames(userName, repoName, Seq(Role.ADMIN)).contains(x.userName) =>
-          action(repository)
-        case _ => Unauthorized()
+    withLockIfEnabled {
+      val userName = params("owner")
+      val repoName = params("repository")
+      getRepository(userName, repoName).map { repository =>
+        context.loginAccount match {
+          case Some(x) if x.isAdmin                      => action(repository)
+          case Some(x) if repository.owner == x.userName => action(repository)
+          // TODO Repository management is allowed for only group managers?
+          case Some(x) if getGroupMembers(repository.owner).exists { m =>
+                m.userName == x.userName && m.isManager
+              } =>
+            action(repository)
+          case Some(x) if getCollaboratorUserNames(userName, repoName, Seq(Role.ADMIN)).contains(x.userName) =>
+            action(repository)
+          case _ => Unauthorized()
+        }
+      } getOrElse NotFound()
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
+        action
       }
-    } getOrElse NotFound()
+    } else {
+      action
+    }
   }
 }
 
 /**
  * Allows only signed in users.
  */
-trait UsersAuthenticator { self: ControllerBase =>
+trait UsersAuthenticator { self: ControllerBase & SystemSettingsService =>
   protected def usersOnly(action: => Any) = { authenticate(action) }
   protected def usersOnly[T](action: T => Any) = (form: T) => { authenticate(action(form)) }
 
   private def authenticate(action: => Any) = {
-    context.loginAccount match {
-      case Some(x) => action
-      case None    => Unauthorized()
+    withLockIfEnabled {
+      context.loginAccount match {
+        case Some(x) => action
+        case None    => Unauthorized()
+      }
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
+        action
+      }
+    } else {
+      action
     }
   }
 }
@@ -67,14 +110,26 @@ trait UsersAuthenticator { self: ControllerBase =>
 /**
  * Allows only administrators.
  */
-trait AdminAuthenticator { self: ControllerBase =>
+trait AdminAuthenticator { self: ControllerBase & SystemSettingsService =>
   protected def adminOnly(action: => Any) = { authenticate(action) }
   protected def adminOnly[T](action: T => Any) = (form: T) => { authenticate(action(form)) }
 
   private def authenticate(action: => Any) = {
-    context.loginAccount match {
-      case Some(x) if (x.isAdmin) => action
-      case _                      => Unauthorized()
+    withLockIfEnabled {
+      context.loginAccount match {
+        case Some(x) if (x.isAdmin) => action
+        case _                      => Unauthorized()
+      }
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
+        action
+      }
+    } else {
+      action
     }
   }
 }
@@ -82,83 +137,131 @@ trait AdminAuthenticator { self: ControllerBase =>
 /**
  * Allows only guests and signed in users who can access the repository.
  */
-trait ReferrerAuthenticator { self: ControllerBase & RepositoryService & AccountService =>
+trait ReferrerAuthenticator { self: ControllerBase & RepositoryService & AccountService & SystemSettingsService =>
   protected def referrersOnly(action: RepositoryInfo => Any) = { authenticate(action) }
   protected def referrersOnly[T](action: (T, RepositoryInfo) => Any) = (form: T) => { authenticate(action(form, _)) }
 
   private def authenticate(action: RepositoryInfo => Any) = {
-    val userName = params("owner")
-    val repoName = params("repository")
-    getRepository(userName, repoName).map { repository =>
-      if (isReadable(repository.repository, context.loginAccount)) {
-        action(repository)
-      } else {
-        Unauthorized()
+    withLockIfEnabled {
+      val userName = params("owner")
+      val repoName = params("repository")
+      getRepository(userName, repoName).map { repository =>
+        if (isReadable(repository.repository, context.loginAccount)) {
+          action(repository)
+        } else {
+          Unauthorized()
+        }
+      } getOrElse NotFound()
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
+        action
       }
-    } getOrElse NotFound()
+    } else {
+      action
+    }
   }
 }
 
 /**
  * Allows only signed in users who have read permission for the repository.
  */
-trait ReadableUsersAuthenticator { self: ControllerBase & RepositoryService & AccountService =>
+trait ReadableUsersAuthenticator { self: ControllerBase & RepositoryService & AccountService & SystemSettingsService =>
   protected def readableUsersOnly(action: RepositoryInfo => Any) = { authenticate(action) }
   protected def readableUsersOnly[T](action: (T, RepositoryInfo) => Any) = (form: T) => {
     authenticate(action(form, _))
   }
 
   private def authenticate(action: RepositoryInfo => Any) = {
-    val userName = params("owner")
-    val repoName = params("repository")
-    getRepository(userName, repoName).map { repository =>
-      if (isReadable(repository.repository, context.loginAccount) || !repository.repository.isPrivate) {
-        action(repository)
-      } else {
-        Unauthorized()
+    withLockIfEnabled {
+      val userName = params("owner")
+      val repoName = params("repository")
+      getRepository(userName, repoName).map { repository =>
+        if (isReadable(repository.repository, context.loginAccount) || !repository.repository.isPrivate) {
+          action(repository)
+        } else {
+          Unauthorized()
+        }
+      } getOrElse NotFound()
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
+        action
       }
-    } getOrElse NotFound()
+    } else {
+      action
+    }
   }
 }
 
 /**
  * Allows only signed in users who have write permission for the repository.
  */
-trait WritableUsersAuthenticator { self: ControllerBase & RepositoryService & AccountService =>
+trait WritableUsersAuthenticator { self: ControllerBase & RepositoryService & AccountService & SystemSettingsService =>
   protected def writableUsersOnly(action: RepositoryInfo => Any) = { authenticate(action) }
   protected def writableUsersOnly[T](action: (T, RepositoryInfo) => Any) = (form: T) => {
     authenticate(action(form, _))
   }
 
   private def authenticate(action: RepositoryInfo => Any) = {
-    val userName = params("owner")
-    val repoName = params("repository")
-    getRepository(userName, repoName).map { repository =>
-      if (isWritable(repository.repository, context.loginAccount)) {
-        action(repository)
-      } else {
-        Unauthorized()
+    withLockIfEnabled {
+      val userName = params("owner")
+      val repoName = params("repository")
+      getRepository(userName, repoName).map { repository =>
+        if (isWritable(repository.repository, context.loginAccount)) {
+          action(repository)
+        } else {
+          Unauthorized()
+        }
+      } getOrElse NotFound()
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
+        action
       }
-    } getOrElse NotFound()
+    } else {
+      action
+    }
   }
 }
 
 /**
  * Allows only the group managers.
  */
-trait GroupManagerAuthenticator { self: ControllerBase & AccountService =>
+trait GroupManagerAuthenticator { self: ControllerBase & AccountService & SystemSettingsService =>
   protected def managersOnly(action: => Any) = { authenticate(action) }
   protected def managersOnly[T](action: T => Any) = (form: T) => { authenticate(action(form)) }
 
   private def authenticate(action: => Any) = {
-    context.loginAccount match {
-      case Some(x) if x.isAdmin                      => action
-      case Some(x) if x.userName == request.paths(0) => action
-      case Some(x) if getGroupMembers(request.paths(0)).exists { member =>
-            member.userName == x.userName && member.isManager
-          } =>
+    withLockIfEnabled {
+      context.loginAccount match {
+        case Some(x) if x.isAdmin                      => action
+        case Some(x) if x.userName == request.paths(0) => action
+        case Some(x) if getGroupMembers(request.paths(0)).exists { member =>
+              member.userName == x.userName && member.isManager
+            } =>
+          action
+        case _ => Unauthorized()
+      }
+    }
+  }
+  
+  private def withLockIfEnabled[T](action: => T): T = {
+    if (loadSystemSettings().basicBehavior.authenticatorLockEnabled) {
+      AuthenticatorLock.lock.synchronized {
         action
-      case _ => Unauthorized()
+      }
+    } else {
+      action
     }
   }
 }
