diff --git a/contrib/gitbucket.conf b/contrib/gitbucket.conf
index b500d390..877815ad 100644
--- a/contrib/gitbucket.conf
+++ b/contrib/gitbucket.conf
@@ -22,7 +22,8 @@ function isMac {
 GITBUCKET_HOST=0.0.0.0
 
 # Other Java option
-GITBUCKET_JVM_OPTS=-Dmail.smtp.starttls.enable=true
+# To enable the authenticator lock for resolving concurrency issues under high load, add -Dgitbucket.authenticator.lock=true
+GITBUCKET_JVM_OPTS="-Dmail.smtp.starttls.enable=true"
 
 # Data directory, holds repositories
 GITBUCKET_HOME=/var/lib/gitbucket
diff --git a/src/main/scala/gitbucket/core/util/Authenticator.scala b/src/main/scala/gitbucket/core/util/Authenticator.scala
index ba838d27..b088cacb 100644
--- a/src/main/scala/gitbucket/core/util/Authenticator.scala
+++ b/src/main/scala/gitbucket/core/util/Authenticator.scala
@@ -103,21 +103,29 @@ trait ReferrerAuthenticator { self: ControllerBase & RepositoryService & Account
  * Allows only signed in users who have read permission for the repository.
  */
 trait ReadableUsersAuthenticator { self: ControllerBase & RepositoryService & AccountService =>
+  private val authLock = new java.util.concurrent.locks.ReentrantLock()
+
   protected def readableUsersOnly(action: RepositoryInfo => Any) = { authenticate(action) }
   protected def readableUsersOnly[T](action: (T, RepositoryInfo) => Any) = (form: T) => {
     authenticate(action(form, _))
   }
 
   private def authenticate(action: RepositoryInfo => Any) = {
-    val userName = params("owner")
-    val repoName = params("repository")
-    getRepository(userName, repoName).map { repository =>
-      if (isReadable(repository.repository, context.loginAccount) || !repository.repository.isPrivate) {
-        action(repository)
-      } else {
-        Unauthorized()
-      }
-    } getOrElse NotFound()
+    val useLock = gitbucket.core.util.ConfigUtil.getConfigValue[Boolean]("gitbucket.authenticator.lock").getOrElse(false)
+    if (useLock) authLock.lock()
+    try {
+      val userName = params("owner")
+      val repoName = params("repository")
+      getRepository(userName, repoName).map { repository =>
+        if (isReadable(repository.repository, context.loginAccount) || !repository.repository.isPrivate) {
+          action(repository)
+        } else {
+          Unauthorized()
+        }
+      } getOrElse NotFound()
+    } finally {
+      if (useLock) authLock.unlock()
+    }
   }
 }
 
