diff --git a/build.sbt b/build.sbt
index 867377eb..45022f70 100644
--- a/build.sbt
+++ b/build.sbt
@@ -62,7 +62,8 @@ libraryDependencies ++= Seq(
   "javax.servlet"                   % "javax.servlet-api"         % "3.1.0"         % "provided",
   "junit"                           % "junit"                     % "4.13.2"        % "test",
   "org.scalatra"                   %% "scalatra-scalatest-javax"  % ScalatraVersion % "test",
-  "org.mockito"                     % "mockito-core"              % "5.23.0"        % "test",
+  "org.specs2"                    %% "specs2-core"               % "4.20.6"        % "test",
+  "org.mockito"                    % "mockito-core"              % "5.23.0"        % "test",
   "org.testcontainers"              % "testcontainers-mysql"      % "2.0.5"         % "test",
   "org.testcontainers"              % "testcontainers-postgresql" % "2.0.5"         % "test",
   "net.i2p.crypto"                  % "eddsa"                     % "0.3.0",
